build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java

```diff
@@ -1,56 +1,71 @@
 import org.jfrog.build.extractor.docker.types.DockerLayer;
 import org.jfrog.build.extractor.docker.types.DockerLayers;
 import org.jfrog.build.extractor.packageManager.PackageManagerExtractor;
 
+import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static org.jfrog.build.extractor.docker.DockerUtils.createMapper;
 import static org.jfrog.build.extractor.packageManager.PackageManagerUtils.createArtifactoryClientConfiguration;
 
 public class BuildDockerCreate extends PackageManagerExtractor {
+    private final ArrayListMultimap<String, String> artifactProperties;
     private final ArtifactoryManagerBuilder artifactoryManagerBuilder;
+    private final ImageFileType imageFileType;
+    private final String sourceRepo;
+    private final String imageFile;
     private final Log logger;
-    private final String kanikoImageFile;
-    private final String targetRepository;
-    private final List<Module> modulesList = new ArrayList<>();
-    private final ArrayListMultimap<String, String> artifactProperties;
 
+    enum ImageFileType {
+        KANIKO,
+        JIB
+    }
 
     /**
      * @param artifactoryManagerBuilder - Artifactory manager builder.
-     * @param targetRepository          - The repository it'll deploy to.
-     * @param kanikoImageFile           - Image file to add.
+     * @param sourceRepo                - The repository it'll resolve from.
+     * @param imageFileType             - The input imageFile format - JIB or Kaniko
+     * @param imageFile                 - Image file to add.
      * @param logger                    - The logger.
      * @param artifactProperties        - Properties to be attached to the docker layers deployed to Artifactory.
      */
-    public BuildDockerCreate(ArtifactoryManagerBuilder artifactoryManagerBuilder,
-                      String kanikoImageFile, ArrayListMultimap<String, String> artifactProperties, String targetRepository, Log logger) {
+    public BuildDockerCreate(ArtifactoryManagerBuilder artifactoryManagerBuilder, String imageFile, ImageFileType imageFileType,
+                             ArrayListMultimap<String, String> artifactProperties, String sourceRepo, Log logger) {
         this.artifactoryManagerBuilder = artifactoryManagerBuilder;
-        this.targetRepository = targetRepository;
-        this.logger = logger;
-        this.kanikoImageFile = kanikoImageFile;
         this.artifactProperties = artifactProperties;
+        this.sourceRepo = sourceRepo;
+        this.imageFileType = imageFileType;
+        this.imageFile = imageFile;
+        this.logger = logger;
     }
 
     /**
-     * Allow running docker push using a new Java process.
+     * Allow running build docker create using a new Java process.
```

Comment:
Allow running build docker create using a new Java process.
-->
Allow creating build-info for a published docker image in a new Java process.